spy.go

```diff
@@ -42,3 +103,65 @@ func Execute(w io.Writer, commandLine string) error {
 
 	return nil
 }
+
+// NewSession creates and returns a new shell session with a file
+func NewSession(filepath string, opts ...Option) (session, error) {
```

Comment:
Since the methods on `session` are pointer methods, it makes sense to return a `*session` here, doesn't it? Then we can simply return `nil` for the error cases.